system/mp/mpcomm.cpp

```diff
@@ -2237,6 +2237,37 @@ int CMPConnectThread::run()
                 connectHdr.id[0].get(_remoteep);
                 connectHdr.id[1].get(hostep);
 
+                // HPCC-28125 - if remoteep IP != peerEp IP then log warning - do we change remoteep ?
+                //              if hostep != listen ep then log warning - do we change hostep ?
+                // **VERY CONCERNED** about pods and local IPs and Network Address Translation
+
+                if (!_remoteep.ipequals(peerEp))
+                {
+                    StringBuffer rIp, pIp;
+                    _remoteep.getIpText(rIp);
+                    peerEp.getIpText(pIp);
+                    PROGLOG("MP: WARNING: remote client address (%s:%u) != peer address (%s:%u)", rIp.str(), _remoteep.port, pIp.str(), peerEp.port);
+                    _remoteep.set(peerEp);
+                    connectHdr.id[0].set(_remoteep);
+                    // should we sock->close() and continue ?
+                }
+
+                SocketEndpoint listenEp;
+                listensock->getEndpoint(listenEp);
+                unsigned short lPort = listenEp.port;
+                IpAddress myLocalIp;
+                GetHostIp(myLocalIp);
+                if (!hostep.ipequals(myLocalIp))
```

Comment:
this is going to be the norm. in the cloud I think (haven't confirmed), and not that unexpected in general.

I can see why the remoteep vs peerEp is a problem that will cause clashes etc.
That needs fixing and I think that fix looks correct.
But I'm not sure what problems the client's notion of what the Dali IP is, vs what Dali itself knows its IP is, is problematic?